### DISCUSSION TITLE
Ollama Model Pull Error in Huggingface Space

### DISCUSSION THREAD

**Discussion by williamzebrowskI** [open-webui/open-webui#4440] [General]
I have hosted open-web UI on a huggingface space but having some difficultly downloading a model via Ollama from the Seetings/Model page.  

I receive this Error in the logs:
`ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Name or service not known]`

If this functionality is even possible in a Huggingface Space, I'm assuming there may be some configuration I have to one of the Docker/docker-composes?

Apologize for the vagueness of this 'Issue'.  It's more of a question if it's possible.

Thank you.

**Comment by justinh-rahb** [comment]
This functionality is not possible in a HF Space unless you use an external-hosted Ollama instance accessed via an HTTP tunnel. For deployment to HF Space, your most convenient options are external OpenAI-compatible APIs (or others via Functions adapters).
